hazelcast/src/main/java/com/hazelcast/sql/SqlQuery.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql;
+
+import com.hazelcast.config.SqlConfig;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Definition of the SQL query.
+ * <p>
+ * This object is mutable. Properties are read once before the execution is started.
+ * Changes to properties do not affect the behavior of already running queries.
+ */
+public class SqlQuery implements Serializable {
```

Comment:
is it a good idea to use Java serialization here? Given this can be sent between different minor versions

Reply:
Agree. This was not part of the original proposal. I added it to provide some convenience to users but didn't think about compatibility. 
Removed now.